content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+
+```bash
+$ pwd
+.../cyf/week-4-example
+```
+
+{{</note>}}
+
+To start, we need to create a file called `package.json` in our project.
+
+> A `package.json` is used to store useful information about our current project, like the name, description, version etc.
+
+We can create a `package.json` file and add the following to it:
+
+package.json
+
+```json
+{
+  "name": "week-4-test-example"
+}
+```
+
+Anyone using this project can now check the `package.json` to get the name of the project.
+We can also add a description:
+
+```json
+{
+  "name": "week-4-test-example",
+  "description": "An example application showing how to write tests using the jest framework"
+}
+```
+
+We can continue adding more information about our project as the project grows.
+For now we can double check we only have a `package.json` in our project:
+
+```bash
+$ ls
+package.json
+```
+
+### 📦 Using packages
+
+When writing software, we need to continually make use of **packages** written by other developers.
+
+A **package** is a set of programs which are grouped together to provide some functionality.
+
+Different programming languages give developers different ways of accessing packages for use in their code.
+
+In JavaScript, we can use **npm**.
+
+> **npm** is a tool used for accessing packages that can be downloaded and used to help developers write applications. The **npm registry** is an online collection of packages that we can download using **npm** commands.
+
+### 🃏 Installing Jest
+
+Our next step will be to figure out how to install the Jest package on our machine.
+
+We can find out more about the Jest framework from the [documentation online](https://jestjs.io/docs/getting-started).
+
+![jest-install](jest-install.png)
+
+In the **Getting started** section of the documentation, Jest gives us the following command:
+
+```bash
+npm install --save-dev jest
+```
+
+- `npm` is an online database of packages, like Jest, that we can fetch and install on to our machines. It also provides us with
+- in the terminal, we can use the `npm` command to install packages on the
+
+`install` - download a package on to our machine.
+
+- `--save-dev` - this indicates that Jest is a package we only use when our application is being developed.
+
+- `jest` - this is the name of the package we want to install on our machine
+
+So overall we can think of this command as saying:
+_"Please go to the npm database, find the Jest package and install it on my local machine"_
+
+Let's execute this command in the same directory as the `package.json`.
+To double check this we can run `pwd`:
+
+```bash
+$ pwd
+.../cyf/week-4-example
+```
+
+`pwd` is telling us we're in the `week-4-example` directory.
+
+We need to double check the `package.json` is also there too.
+
+```bash
+$ ls
+package.json
+```
+
+Now we can execute the command
+
+```bash
+npm install --save-dev jest
+```
+
+Our project structure will now look as follows:
+
+```raw
+week-4-example
+├── get-ordinal-number.test.js
```

Comment:
I'm following along at home, and I don't have a `get-ordinal-number.test.js`, but I do have an unexplained `node_modules`...